### DISCUSSION TITLE
Error patching node with cloud ip addresses

### DISCUSSION THREAD

**Discussion by AndrewDevelopment** [k3s-io/k3s#8980] [Q&A]
I encounter on an error patching node in cluster, i setting up three server, they are all master,worker node.
In order to let my ap pods be able to request service, i setup a virtual ip by using keepalived, all request come from this ip to my ap pods.

virtualip: 192.168.1.13
server1:192.168.1.10
server2: 192.168.1.11
server3: 192.168.1.12
Below is the error message from the server3.

4415 node_controller.go:363] Error patching node with cloud ip addresses = [failed to patch status “{"metadata":{"$setElementOrder/finalizers":["wrangler.cattle.io/node","wrangler.cattle.io/managed-etcd-controller"]},"status":{"addresses":[{"
address":"192.168.1.12","type":"InternalIP"},{"address":"192.168.1.12","type":"InternalIP"},{"address":"servertry3","type":"Hostname"},{"$patch":"replace"}]}}” for node “servertry3”: Node “servertry3” is invalid: status.addresses[1]: Duplicate value: core.NodeAddress{Type:“InternalIP”, Address:“192.168.1.12”}]

**Comment by brandond** [comment]
Can you provide exact steps to reproduce this? What flags are you starting k3s with? It is unclear what specifically you are doing and where/when you see the error.

Normally many of these questions would be answered by creating an issue and filling out the issue template.

**Reply by AndrewDevelopment** [reply]
hi brandond:
                   this question will be transfer to issue, i not sure how to reproduce the step to show the error message, i see this error by logs on /var/log/syslog. 
                   the virtual ip is created by using keepalived, below is the configuration set on three server:
```
    vrrp_instance VI_1 {
    state BACKUP
    interface eno1
    virtual_router_id 1
    priority 100
    advert_int 1
    nopreempt
    authentication {
        auth_type AH
        auth_pass testserver1234
    }

    unicast_src_ip 192.168.1.10
    unicast_peer {
        192.168.1.11
        192.168.1.12
    }

    virtual_ipaddress {
        192.168.1.13
    }
}
```

**Reply by brandond** [reply]
Are you using the keepalived vip as the --node-ip or something weird like that?

**Reply by AndrewDevelopment** [reply]
the vip is only use for communication with the service pods in cluster such as minio and postgres, but also from ap from other too.

**Reply by brandond** [reply]
What does your K3s config look like? For some reason it's complaining that you've got the same IP listed twice somewhere.

**Reply by AndrewDevelopment** [reply]
i have look through the ip, but not found anything yet, do you have recommended way to check it?

**Reply by brandond** [reply]
`kubectl get node -o yaml`
